From 0d924cc74caf41ca458abfaf66dac7bd1de1e97f Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Wed, 11 Dec 2013 21:26:24 +0100
Subject: [PATCH 325/327] MEDIUM: checks: enable agent checks even if health
 checks are disabled

The agent is able to retrieve some weight information from the server
and will eventually be able to force the server into maintenance mode.
It doesn't seem logical to have it depend on the health check being
configured, as for some servers it might very well make sense to only
fetch the weight from the server's load regardless of the health.

So let's stop disabling the agent checks when health checks are disabled.
---
 src/checks.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/checks.c b/src/checks.c
index 9b00062..9fdf060 100644
--- a/src/checks.c
+++ b/src/checks.c
@@ -1507,9 +1507,7 @@ static struct task *process_chk(struct task *t)
 		 * stopped, the server should not be checked or the check
 		 * is disabled.
 		 */
-		if (!(check->state & CHK_ST_ENABLED) ||
-		    !(s->check.state & CHK_ST_ENABLED) ||
-		    (s->check.state & CHK_ST_PAUSED) ||
+		if (((check->state & (CHK_ST_ENABLED | CHK_ST_PAUSED)) != CHK_ST_ENABLED) ||
 		    s->proxy->state == PR_STSTOPPED)
 			goto reschedule;
 
-- 
1.8.1.5

